core/state/database.go

```diff
@@ -238,19 +238,45 @@ func (db *cachingDB) purgeLoop() {
 
 // OpenTrie opens the main account trie at a specific root hash.
 func (db *cachingDB) OpenTrie(root common.Hash) (Trie, error) {
+	//try cache first
+	if db.accountTrieCache != nil {
+		if tr, ok := db.accountTrieCache.Get(root); ok {
+			return tr.(*trie.SecureTrie), nil
+		}
+	}
+
 	tr, err := trie.NewStateTrie(trie.StateTrieID(root), db.triedb)
 	if err != nil {
 		return nil, err
 	}
+
+	//update cache
+	db.CacheAccount(root, tr)
```

Comment:
no need to cache here, it will be cached when `statedb.commit`

Reply:
As far as I understand, `StateDB.Commit()` only commits dirty modified entries; if an entry isn't modified, it won't be cached by `StateDB.Commit()`.

Therefore, I think it makes sense to explicitly cache entries here.